packages/build/src/plugins/resolve.js

```diff
@@ -55,17 +55,17 @@ const resolvePluginPath = async function ({
     return pluginOptions
   }
 
-  const packageNameA = resolvePackagePath(packageName)
+  const localPackageName = normalizeLocalPackageName(packageName)
 
   // Local plugins
-  if (packageNameA.startsWith('.')) {
-    const localPath = await tryLocalPath(packageNameA, buildDir)
+  if (localPackageName.startsWith('.')) {
+    const localPath = await tryLocalPath(localPackageName, buildDir)
     return { ...pluginOptions, pluginPath: localPath, loadedFrom: 'local' }
   }
 
   // Plugin already installed in the project, most likely either local plugins,
   // or external plugins added to `package.json`
-  const manualPath = await tryResolvePath(packageNameA, buildDir)
+  const manualPath = await tryResolvePath(packageName, buildDir)
```

Comment:
Plugins in `package.json` and in the build image never usage local paths, so we clearly distinguish `localPackageName` and `packageName` here.